src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -194,6 +218,21 @@ public void OnResponseHeadersStart()
                 }
             }
 
+            public void OnResponse100Continue()
+            {
+                // This is called when we receive complete set of headers with status 100 Continue.
+                // We need to rest state and wait for another final response.
+                lock (SyncObject)
+                {
+                    if (_state != StreamState.ExpectingData)
+                    {
+                        throw new Http2ProtocolException(Http2ProtocolErrorCode.ProtocolError);
```

Comment:
@stephentoub Did a recent PR which improved error message and the exceptions.  Is this another place that could benefit from that? I.e. instead of just having `Http2ProtocolErrorCode.ProtocolError` it could include other details in a string or something with the specific reason "expecting more data from stream" (or something like that).

Reply:
I probably should have done that in general when updating the Htp2ProtocolErrorCode stuff.  If you don't want to do it as part of this PR, I can take a separate pass through as well.

Reply:
I made update to include more info. I can also make separate PR if preferable.